supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -289,8 +294,9 @@ object TesAsyncBackendJobExecutionActor {
         val tesVmCostData = for {
           responseLogs <- t.logs
           startTime <- responseLogs.headOption.map(_.start_time)
+          endTime <- responseLogs.headOption.map(_.end_time)
```

Comment:
I think that this line _might_ cause issues. There's some somewhat involved polling logic happening. 

tl;dr I don't think we want to require that an `endTime` exist in order to establish `TesVmCostData`, as the presence of that object currently controls Cromwell's polling behavior

My understanding of the polling behavior:
- Cromwell submits a task to TES and polls every once in awhile.
- Cromwell tries to be efficient and not ask for more data than is necessary, and not ask for the cost data after it has been successfully obtained once. This  gets us cost data as soon as it's available, which we can then begin to show on the front end
- Because TES can't send us the end time until the task actually finishes, this change is going to make cromwell do the 'big poll' for the duration of the entire task. 

tl;dr - I don't think we want to require that the `endTime` be known for us to construct a `TesVmCostData`, as the presence of that object is how we control what we poll for.

Reply:
I wonder if we can accomplish everything desired for bard in the `onTaskComplete` function. We'll know the start time for sure, and we can either use the current time for the end time (not 100% accurate, but maybe close enough?) or query TES one last time to ask it for the end time.